controllers/databasecluster_controller.go

```diff
@@ -659,6 +656,49 @@ func (r *DatabaseClusterReconciler) initWatchers(controller *builder.Builder) {
 		}),
 		builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
 	)
+
+	controller.Watches(
+		&psmdbv1.PerconaServerMongoDB{},
+		handler.EnqueueRequestsFromMapFunc(func(_ context.Context, obj client.Object) []reconcile.Request {
+			return []reconcile.Request{
+				{
+					NamespacedName: types.NamespacedName{
+						Name:      obj.GetName(),
+						Namespace: obj.GetNamespace(),
+					},
+				},
+			}
+		}),
+		builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+	)
+	controller.Watches(
+		&pgv2.PerconaPGCluster{},
+		handler.EnqueueRequestsFromMapFunc(func(_ context.Context, obj client.Object) []reconcile.Request {
+			return []reconcile.Request{
+				{
+					NamespacedName: types.NamespacedName{
+						Name:      obj.GetName(),
+						Namespace: obj.GetNamespace(),
+					},
+				},
+			}
+		}),
+		builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+	)
+	controller.Watches(
+		&pxcv1.PerconaXtraDBCluster{},
+		handler.EnqueueRequestsFromMapFunc(func(_ context.Context, obj client.Object) []reconcile.Request {
+			return []reconcile.Request{
+				{
+					NamespacedName: types.NamespacedName{
+						Name:      obj.GetName(),
+						Namespace: obj.GetNamespace(),
+					},
+				},
+			}
+		}),
+		builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+	)
```

Comment:
Can be simplified using `EnqueueRequestForObject` (it achieves the same thing)
```suggestion
	controller.Watches(
		&psmdbv1.PerconaServerMongoDB{},
		&handler.EnqueueRequestForObject{},
		builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
	)
	controller.Watches(
		&pgv2.PerconaPGCluster{},
		&handler.EnqueueRequestForObject{},
		builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
	)
	controller.Watches(
		&pxcv1.PerconaXtraDBCluster{},
		&handler.EnqueueRequestForObject{},
		builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
	)

```